config/clusters/cloudbank/common.values.yaml

```
@@ -1,5 +1,12 @@
 jupyterhub:
   singleuser:
+    cpu:
+      # Each node has about 8 CPUs total, and if we limit users to no more than
+      # 4, no single user can take down a full node by themselves. We have to
+      # set the guarantee to *something*, otherwise it is set to be equal
+      # to the limit!
+      limit: 4
+      guarantee: 0.01
```

I'm not fully confident, but I think we ought to make the guarantee high enough to startup, and I think ~5% of a CPU may be required. I've seen user pods starting up use about that much CPU.

```suggestion
      guarantee: 0.05
```

---

5% of a CPU in gaurantee would allow for up to about 80 pods on a 4 CPU node.

If memory requests are 0.5 GB per user, they would still run short by memory before CPU with 5% on a 4 CPU / 32 GB ram node.

On a 8 CPU or larger node, they would run out of the total number of pods allowed for a node before the 5% of a CPU guaranteed became a problem.

---

Oh, it is default to 0.05 in basehub: https://github.com/2i2c-org/infrastructure/blob/e1d2bd3ad79a9fe276d13a93c0e174710c1b5fb6/helm-charts/basehub/values.yaml#L291-L309

---

@consideRatio I've removed the guarantee and added a comment on why.